UI/AppKit/Interface/LadybirdWebView.mm

```diff
@@ -1072,8 +1072,78 @@ - (void)keyDown:(NSEvent*)event
         return;
     }
 
-    auto key_event = Ladybird::ns_event_to_key_event(Web::KeyEvent::Type::KeyDown, event);
-    m_web_view_bridge->enqueue_input_event(move(key_event));
+    [self interpretKeyEvents:@[ event ]];
+}
+
+- (void)insertText:(id)string replacementRange:(NSRange)replacementRange
+{
+    NSEvent* event = [NSApp currentEvent];
+    if (event && event.type == NSEventTypeKeyDown) {
+        auto key_event = Ladybird::ns_event_to_key_event(Web::KeyEvent::Type::KeyDown, event);
+        m_web_view_bridge->enqueue_input_event(move(key_event));
+    }
+}
+
+- (void)doCommandBySelector:(SEL)selector
+{
+    NSEvent* event = [NSApp currentEvent];
+    if (event && event.type == NSEventTypeKeyDown) {
+        auto key_event = Ladybird::ns_event_to_key_event(Web::KeyEvent::Type::KeyDown, event);
+        m_web_view_bridge->enqueue_input_event(move(key_event));
+    }
+}
+
+- (BOOL)acceptsFirstResponder
```

Comment:
(Though some of these don't seem to be `NSTextInputClient` thus should be in their own mark - looks like `NSResponder` here)